### DISCUSSION TITLE
Calling $this->validate() results in "Call to a member function with() on null"

### DISCUSSION THREAD

**Discussion by zawilliams** [livewire/livewire#7687] [7. Bugs (archived)]
### Livewire version

v3.3.5

### Laravel version

v10.39.0

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

I've created a Wirebox to demo this problem: [wirebox.app/b/47r2z](https://wirebox.app/b/47r2z)

Currently I'm validating a malformed date on line 11:

```php
public string $start = '2024-0101',
```
If you change the date to `2024-01-01`, everything works.

As far as I can tell, everything is setup correctly. I noticed #6656 and originally commented in there, but it is seemingly a different issue even though the error is the same.

Any thoughts here? Is this a bug?

### Repository or Wirebox URL

https://wirebox.app/b/47r2z

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@zawilliams one thing to note, you shouldn't be using a `__construct` with Livewire as it's not considered part of the public API instead you should define properties directly on the component, and if you need to initialise, you should use `mount()`.

The problem you are currently facing, is the validation exception is being thrown in the render method, but normally validation happens before render, so validation messages can be displayed in render. So I would recommend using a computed property instead:

```php
use App\Models\Event;
use Livewire\Attributes\Computed;
use Livewire\Component;

class Events extends Component
{
    // Malformed date:
    public string $start = '2024-0101';
    // Correctly formatted date:
    // public string $start = '2024-01-01',
    public string $end = '2024-01-31';

    #[Computed()]
    public function events()
    {
        $validated = $this->validate([
            'start' => 'date',
            'end' => 'date',
        ]);

        if (count($validated) === 2) {
            return Event::betweenDates($validated['start'], $validated['end'])->get();
        }

        return Event::all();
    }

    public function render()
    {
        return view('livewire.main');
    }
}
```

And you then need to update your view to reference the computed property by using `$this->events` instead of just `$events`.

```blade
<div class="flex flex-col w-full h-full items-center justify-center text-gray-300 px-16">
    @foreach ($this->events as $event)
        <div>{{ $event->name }}</div>
    @endforeach
</div>
```

Hope this helps!

**Reply by zawilliams** [reply]
@joshhanley ah yes, got it, thanks! Good call about `__construct` too with Livewire. Forgot about that.

Ok, so I've updated a fork of the Wirebox and am now seeing a `ValidationException`: https://wirebox.app/b/x6red

I was assuming this would just silently fail and if I wanted to, I could use `@error('start'){{ $message }}@enderror` to show the validation error. If I switch my `APP_ENV` to `production`, I just get a `500`.

Do I have an assumption wrong here or am I just missing something?

**Reply by zawilliams** [reply]
Tried a couple more things and am still seeing the same thing. Will take another look tomorrow to see if I'm missing something in the docs.

**Comment by zawilliams** [comment]
I found a related issue from Livewire 1, which looks to be similar to what I'm experiencing/trying to do here: https://github.com/livewire/livewire/issues/1265

Specifically:

> Laravel Livewire doesn't catch the ValidationException like in normal actions and shows the frontend debugging window.

If this is indeed the same issue but with Livewire 3, and this is in fact, not a bug... I'm wondering how to validate on page load for a query. Specifically, the values in my scenario would be coming from a query param. If someone tweaks a date in the query param to an incorrect format (without validation), the date gets passed to the query scope and the app returns a 500 in production.

Just trying to understand how I can avoid that, or wondering if there is a different approach I should be taking. Putting the validation before the query builder seems to be the right place to put it.
